Movied/Movied/Scenes/Home/HomeViewController.swift

```diff
@@ -1,9 +1,215 @@
 import UIKit
 
 final class HomeViewController: UIViewController {
-
+    @IBOutlet private weak var homeTitle: UILabel!
+    @IBOutlet private weak var movieListCollectionView: UICollectionView!
+    @IBOutlet private weak var sortedByTextField: UITextField!
+    
+    private var allMovies: [Movie] = []
+    private var sortType: SortType = .newestToOldest
+    private var pageNumber = 1
+    
+    private let refreshControl = UIRefreshControl()
+    private let pickerView = UIPickerView()
+    private var refreshFooterView: RefreshFooterView?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        showIndicator(seconds: 2)
+        configView()
+        fetchMovies()
+    }
+    
+    //MARK: - Fetch Data
+    private func fetchMovies() {
+        pageNumber = 1
+        APIService.shared.getDiscoverMovies(page: pageNumber,
+                                            sortType: sortType) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let movieList):
+                DispatchQueue.main.async {
+                    self.allMovies = movieList.results
+                    self.movieListCollectionView.reloadData()
+                }
+            case .failure(let error):
+                print(error.rawValue)
+            }
+        }
+    }
+    
+    private func loadMoreMovies() {
+        pageNumber += 1
+        APIService.shared.getDiscoverMovies(page: pageNumber,
+                                            sortType: sortType) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let movieList):
+                sleep(3)
```

Comment:
why sleep app?

Reply:
Because getting API is so fast, so I sleep for 3 seconds for better UX. But I've changed into asyncAfter.